etcdserver/server.go

```diff
@@ -1479,11 +1484,24 @@ func (s *EtcdServer) restoreAlarms() error {
 }
 
 func (s *EtcdServer) getAppliedIndex() uint64 {
-	return atomic.LoadUint64(&s.appliedIndex)
+	s.appliedIndexMu.Lock()
+	defer s.appliedIndexMu.Unlock()
+	return s.appliedIndex
 }
 
 func (s *EtcdServer) setAppliedIndex(v uint64) {
-	atomic.StoreUint64(&s.appliedIndex, v)
+	s.appliedIndexMu.Lock()
+	defer s.appliedIndexMu.Unlock()
+	s.appliedIndex = v
+	s.appliedIndexCond.Broadcast()
```

Comment:
@heyitsanthony We can batch more entries applied for the bcast here. we could optimize it later once we have bench data.